src/visitors/rewrite_visitor.py

```diff
@@ -198,6 +197,24 @@ def inv_2_rule_3():
 
         return {**inverse_1, **inverse_2}
 
+    def visit_string_concatenation1_1(self, operator: StringConcatenation1_1):
+        return self._visit_string_concatenation(operator, 1, 1)
+
+    def visit_string_concatenation1_2(self, operator: StringConcatenation1_2):
+        return self._visit_string_concatenation(operator, 1, 2)
+
+    def visit_string_concatenation1_3(self, operator: StringConcatenation1_3):
+        return self._visit_string_concatenation(operator, 1, 3)
+
+    def visit_string_concatenation2_1(self, operator: StringConcatenation2_1):
+        return self._visit_string_concatenation(operator, 2, 1)
+
+    def visit_string_concatenation2_2(self, operator: StringConcatenation2_2):
+        return self._visit_string_concatenation(operator, 2, 2)
+
+    def visit_string_concatenation2_3(self, operator: StringConcatenation2_3):
+        return self._visit_string_concatenation(operator, 2, 3)
```

Comment:
Can we just do this normally? I know it's unnecessary code duplication, but I think I'd prefer writing these inverses in the same way as all the others instead of just passing indices.

Reply:
This is more debatable, but I also think just numbering them 1 through 6 would be better. I really don't like the underscore in the CamelCase.

Reply:
I am not sure, my point here is:
* we avoid code duplication as you noticed
* by naming the operators this way and having the function which handles the rewriting of the string concatenation operator, the mapping between operators and rewriting rules is clear
* all the rewriting rules for str.++ are grouped and it's easier to avoid mistakes as duplication of the same combination of rewriting rules as there is one only place where to look for all of them. Also the indeces mapping helps with this